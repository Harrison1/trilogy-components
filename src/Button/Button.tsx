import * as React from 'react';
import styled from '../utils/styled-components';
import GlobalStyles from '../GlobalStyles';
import setRGBA from '../utils/setRGBA'

type ButtonType = 'button' | 'reset' | 'submit' | undefined;

export interface ButtonProps {
  autoFocus?: boolean;
  bold?: boolean;
  ariaLabel?: string;
  background?: string;
  children?: React.ReactNode;
  disabled?: boolean;
  fontColor?: string;
  fontSize?: string;
  name?: string;
  primary?: boolean;
  onClick?: (arg0: any) => void;
  type?: ButtonType;
  value?: string;
}

const TButton = styled(`button`)<ButtonProps>`
  background-color: ${props =>
    props.background ? props.background : '#28c8e5'};
  border: none;
  color: ${props => (props.fontColor ? props.fontColor : '#fff')};
  cursor: pointer;
  font-size: ${props => (props.fontSize ? props.fontSize : '2.4rem')};
  font-weight: ${props => (props.bold ? `700` : `400`)};
  height: 48px;
  width: 100%;
  :hover {
    background-color: ${props => props.background ? setRGBA(props.background) : '#28c8e5BF'};
  }
`;

const noop = () => {};

export const Button = (props: ButtonProps) => (
  <>
    <GlobalStyles />
    <TButton
      ariaLabel={props.ariaLabel ? props.ariaLabel : props.name}
      autoFocus={props.autoFocus}
      background={props.background}
      bold={props.bold}
      fontColor={props.fontColor}
      disabled={props.disabled}
      name={props.name}
      onClick={!props.disabled ? props.onClick : noop}
      type={props.type}
      value={props.value}
      primary={props.primary ? true : false}
      fontSize={props.fontSize}
    >
      {props.children}
    </TButton>
  </>
);
