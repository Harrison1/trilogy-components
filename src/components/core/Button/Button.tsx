import * as React from 'react';
import styled from '../../utils/styled-components';

type ButtonType = 'button' | 'reset' | 'submit' | undefined;

export interface ButtonProps {
  autoFocus?: boolean;
  ariaLabel?: string;
  background?: string;
  children?: React.ReactNode;
  disabled?: boolean;
  fontColor?: string;
  fontSize?: string;
  name?: string;
  primary?: boolean;
  onClick?: () => void;
  type?: ButtonType;
  value?: string;
}

const TButton = styled(`button`)<ButtonProps>`
  background: ${props => (props.background ? props.background : '#28c8e5')};
  border: none;
  color: ${props => (props.fontColor ? props.fontColor : '#fff')};
  cursor: pointer;
  font-size: ${props => (props.fontSize ? props.fontSize : '1rem')};
  padding: 4px;
  &:hover {
    animation-play-state: running;
  }
  &:active,
  &:focus,
  &:visited {
    border-color: #80bdff;
    outline: 0;
    box-shadow: 0 0 0 0.3rem rgba(180, 0, 255, 0.35);
  }
`;

const noop = () => {};

export const Button = (props: ButtonProps) => (
  <TButton
    ariaLabel={props.ariaLabel ? props.ariaLabel : props.name}
    autoFocus={props.autoFocus}
    background={props.background}
    fontColor={props.fontColor}
    disabled={props.disabled}
    name={props.name}
    onClick={!props.disabled ? props.onClick : noop}
    type={props.type}
    value={props.value}
    primary={props.primary ? true : false}
    fontSize={props.fontSize}
  >
    {props.children}
  </TButton>
);
