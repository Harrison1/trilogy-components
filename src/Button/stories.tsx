import * as React from 'react';
import { color, withKnobs } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';
import Button from '.';
import SectionTitle from '../SectionTitle'

const stories = storiesOf('Button', module);

stories.addDecorator(withKnobs);

stories.add('Default', () => (
  <>
    <SectionTitle text='Button' bold />
    {/* TODO: docs go here */}
    <Button
      background={color('Background Color', '#28c8e5')}
      fontColor={color('Text Color', '#fff')}
    >
      Galaxy
    </Button>
  </>
));
