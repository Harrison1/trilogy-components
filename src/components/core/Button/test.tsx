import * as React from 'react';
import { Button } from './Button';
import * as renderer from 'react-test-renderer';

test('Standard button', () => {
  const component = renderer.create(<Button>Hello Galaxy</Button>);

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
