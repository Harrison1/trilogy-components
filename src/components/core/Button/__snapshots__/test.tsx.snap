// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Standard button 1`] = `
<button
  className="sc-bdVaJa iPqJvl"
>
  Hello Galaxy
</button>
`;
